blocks/collapse/collapse.js

```diff
@@ -1,122 +1,32 @@
-                    </span>
-    
-                    <span class="icon icon-expand-more"></span>
-                  </button>
-                </div>
-              </div>
-            </div>
-
-          </div>
-        </div>
-      </div>
-    </div>
-  `).join('');
-}
-
-function addCollapseListener(collapse) {
-  const expandable = collapse.querySelector('.text-block__expandable-area');
-  expandable.style.transition = 'height 500ms ease-in-out';
-
-  const textBlock = expandable.querySelector('.text');
-  const textBlockButton = expandable.querySelector('.text-block__button');
-
-  collapse.addEventListener('click', (event) => {
-    const toggle = event.target.closest('.button-toggle');
-    if (toggle) {
-      const isExpanded = toggle.classList.contains('button-toggle--expanded');
-      const isCollapsed = toggle.classList.contains('button-toggle--collapsed');
-
-      collapse.classList.toggle('text-block--expanded', !isExpanded);
-      toggle.classList.toggle('button-toggle--expanded', !isExpanded);
-      toggle.classList.toggle('button-toggle--collapsed', !isCollapsed);
-
-      const button = toggle.querySelector('button');
-      const buttonContent = button.querySelector('.button-link__content');
-      const newLabel = isExpanded ? toggle.dataset.collapsedLabel : toggle.dataset.expandedLabel;
-      button.setAttribute('aria-label', newLabel);
-      button.setAttribute('title', newLabel);
-      buttonContent.textContent = newLabel;
+function addCollapseListener(collapseBlock) {
+  const expandableText = collapseBlock.querySelector('div:nth-child(2)');
+  const collapsibleSection = collapseBlock.parentElement.parentElement;
+  const expandButton = collapseBlock.querySelector('div:nth-child(3)');
+  const collapseButton = collapseBlock.querySelector('div:nth-child(4)');
+  const arrowUpElement = document.createElement('div');
+  if (!expandButton && !collapseButton) return;
+  arrowUpElement.classList.add('arrow', 'down');
+  expandButton.appendChild(arrowUpElement);
+  const arrowDownElement = document.createElement('div');
+  arrowDownElement.classList.add('arrow', 'up');
+  collapseButton.appendChild(arrowDownElement);
+
+  expandButton.addEventListener('click', () => {
+    const height = collapseBlock.querySelector('div:nth-child(2) > div').clientHeight
+        + collapseButton.getBoundingClientRect().height
+        + parseFloat(getComputedStyle(expandableText).paddingBottom);
+    expandableText.style.height = `${height}px`;
+    collapsibleSection.classList.remove('collapsed-text');
+    collapsibleSection.classList.add('expanded-text');
+  });
 
-      const height = isCollapsed ? textBlock.getBoundingClientRect().height + textBlockButton.getBoundingClientRect().height + parseFloat(getComputedStyle(expandable).paddingBottom) : '0';
-      expandable.style.height = `${height}px`;
-    }
+  collapseButton.addEventListener('click', () => {
+    expandableText.style.height = '0';
```

Comment:
I'd avoid inline styles and use css classes instead.

Reply:
The issue here is that we need to change the style of the expandable text component when the button is clicked, so from what I see it needs to be in the js code.